### PR TITLE
add documentation to android dsl

### PR DESCRIPTION
**Pull request by estute**
@jibsheet

### CODE REVIEW

**Comment by jibsheet** [comment]
@feanil can you be the second set of eyes who doesn't know DSL as well?

File: mobileApp/jobs/buildWhiteLabelAndroidApps.groovy [hunk 1]
```diff
@@ -3,24 +3,44 @@ package mobile
...
+   - Publish the apk to HockeyApp (a service for hosting & disrtibuting application files)
+   - TODO: Run tests on the generated apk file (linting, unit tests, screenshot tests)
+ Depending on the "release" flag mentioned below, the job will be considered either 'release' or 'debug'
+ Release builds will also use special gradle.properties and keystore files when compiled, to sign
+ the build.
+
+ ==========================================================================================
+ In order to run, the job requires the following be loaded into Jenkins as secrets files:
+   - A secret yml file: This contains a number of jobConfigs, one for each desired Jenkins job
+   - A gradle.properties file: A key-value file used in the gradle build process **
```
Comment (feanil) [inline mobileApp/jobs/buildWhiteLabelAndroidApps.groovy]: You mentioned above that there will be two different sets of (gradle.properties, keystore)  So in total 5 files will be uploaded is that right?  Perhaps we can enumerate both sets. eg.
` A debug gradle.properties file:...`
` A debug keystore file:...`
`A release gradle.properties file:...`
`A release keystore file:...`

**Comment by feanil** [comment]
One issue otherwise generally makes sense to me.

File: mobileApp/jobs/buildWhiteLabelAndroidApps.groovy (hunk 1 above)
Comment (jibsheet) [inline mobileApp/jobs/buildWhiteLabelAndroidApps.groovy]: I think we do need two gradle properties, but there's just no keystore for debug builds (there's a ** footnote below).

**Comment by jibsheet** [comment]
I agree @estute clear up how we differentiate the gradle properties and I think you're good to go.

**Pull request merged**